src/generators.rs

```diff
@@ -551,6 +551,329 @@ pub fn star_graph(
+///     are not specified, then a linear graph containing all the
+///     values in weights list is created.
+///     If number of nodes(rows*cols) is less than length of
+///     weights list, the trailing weights are ignored.
+///     If number of nodes(rows*cols) is greater than length of
+///     weights list, extra nodes with None weight are appended
+///
+/// :returns: The generated star graph
+/// :rtype: PyGraph
+/// :raises IndexError: If neither ``rows`` or ``cols`` and ``weights`` are
+///      specified
+///
+#[pyfunction]
+#[text_signature = "(/, rows=None, cols=None, weights=None)"]
+pub fn grid_graph(
+    py: Python,
+    rows: Option<usize>,
+    cols: Option<usize>,
+    weights: Option<Vec<PyObject>>,
+) -> PyResult<graph::PyGraph> {
+    let mut graph = StableUnGraph::<PyObject, PyObject>::default();
+    if weights.is_none() && (rows.is_none() || cols.is_none()) {
+        return Err(PyIndexError::new_err(
+            "dimensions and weights list not specified",
+        ));
+    }
+
+    let mut rowlen = match rows {
+        Some(rows) => rows,
+        None => 0,
+    };
+
+    let mut collen = match cols {
+        Some(cols) => cols,
+        None => 0,
+    };
+
+    let mut num_nodes = rowlen * collen;
+
+    let nodes: Vec<NodeIndex> = match weights {
+        Some(weights) => {
+            let mut node_list: Vec<NodeIndex> = Vec::new();
+            if num_nodes < weights.len() && rowlen == 0 {
+                collen = weights.len();
+                rowlen = 1;
+                num_nodes = collen;
+            }
+
+            let mut node_cnt = num_nodes;
+
+            for weight in weights {
+                if node_cnt <= 0 {
+                    break;
+                }
+                let index = graph.add_node(weight);
+                node_list.push(index);
+                node_cnt -= 1;
+            }
+            for _i in 0..node_cnt {
+                let index = graph.add_node(py.None());
+                node_list.push(index);
+            }
+            node_list
+        }
+        None => (0..num_nodes).map(|_| graph.add_node(py.None())).collect(),
+    };
+
+    for i in 0..rowlen {
+        for j in 0..collen {
+            if i + 1 < rowlen {
+                graph.add_edge(
+                    nodes[i * collen + j],
+                    nodes[(i + 1) * collen + j],
+                    py.None(),
+                );
+            }
+            if j + 1 < collen {
+                graph.add_edge(
+                    nodes[i * collen + j],
+                    nodes[i * collen + j + 1],
+                    py.None(),
+                );
+            }
+        }
+    }
+    Ok(graph::PyGraph {
+        graph,
+        node_removed: false,
+    })
+}
+
+/// Generate a directed grid graph. The edges propagate towards right and
+///     bottom direction if ``bidirectional`` is ``false``
+///
+//// :param int rows: The number of rows to generate the graph with.
```

Comment:
This should fix the failing docs build I think:

```suggestion
/// :param int rows: The number of rows to generate the graph with.
```